File: src/main/scala/samples/CommentListExample/Notification/Notification.scala
package samples.CommentListExample.Notification

import com.github.ahnfelt.react4s._
import samples.CommentListExample.CommentsStore

case class Notification() extends Component[NoEmit] {
  val commentStore = CommentsStore(this)

  override def render(get: Get): Element = E.div(
    if (get(commentStore).length > 2) {
      E.h1(Text("Too many Comments. Reduce those"))
    } else {
      E.h1()
    }
  )
}

File: react-redux/src/notification/Notification.js
import React, {Component} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

class Notification extends Component {

    constructor() {
        super();

        this._showNotification = this._showNotification.bind(this);
    }


    _showNotification() {
        if(this.props.noOfComments > 2) {
            return <h1>Too many Comments. Reduce those</h1>
        }
    }

    render() {
        return (
            <div className="notification">
                {
                    this._showNotification()
                }
            </div>
        );
    };


}

Notification.propTypes = {
  comments: PropTypes.array
};

const mapStateToProps = (store) => {
  return {
    noOfComments: store.comments.length
  };
};

export default connect(
  mapStateToProps
)(Notification);